### PR TITLE
Rewrite and rename `issue-14698`. `issue-33329` and `issue-107094` `run-make` tests to rmake or ui

### PR DESCRIPTION
**Pull request by Oneirical**
Part of #121876 and the associated [Google Summer of Code project](https://blog.rust-lang.org/2024/05/01/gsoc-2024-selected-projects.html).

try-job: armhf-gnu
try-job: test-various
try-job: aarch64-apple
try-job: x86_64-msvc

### CODE REVIEW

File: tests/run-make/invalid-tmpdir-env-var/rmake.rs [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+use run_make_support::rustc;
+
+// NOTE: This is not a UI test despite its simplicity, as the error message contains a path
+// with some variability that is difficult to normalize
+
+fn main() {
+    rustc()
+        .input("foo.rs")
+        .env("TMP", "fake")
+        .env("TMPDIR", "fake")
```
Comment (jieyouxu) [inline tests/run-make/invalid-tmpdir-env-var/rmake.rs:15]: Suggestion: only set `TMP` for Windows, and `TMPDIR` for Linux/macOS -- they are the respectively platform-specific env vars that influence what [`std::env::temp_dir`](https://doc.rust-lang.org/std/env/fn.temp_dir.html#) returns.

File: tests/run-make/invalid-tmpdir-env-var/rmake.rs [hunk 2]
```diff
@@ -0,0 +1,18 @@
+// When the TMP or TMPDIR variable is set to an invalid or non-existing directory,
+// this used to cause an internal compiler error (ICE). After the addition of proper
+// error handling in #28430, this test checks that the expected message is printed.
+// See https://github.com/rust-lang/rust/issues/14698
```
Comment (jieyouxu) [inline tests/run-make/invalid-tmpdir-env-var/rmake.rs:4]: Suggestion: please specify *which* env vars influences `std::env::temp_dir`:

- `TMPDIR` on Unix (Linux, macOS)
- `TMP` > `TEMP` > `USERPROFILE` > `C:\Windows` on Windows (`>` indicates fallback precedence if the env var on LHS is not set)

**Comment by jieyouxu** [comment]
Thanks!
@bors r+ rollup

**Comment by Oneirical** [comment]
Conflict removed, please re-approve.
@rustbot ready

**Comment by jieyouxu** [comment]
Thanks!

bors-r+

**Pull request merged**